lua/octo/workflow_runs.lua

```diff
@@ -0,0 +1,686 @@
+  end
+
+  vim.api.nvim_buf_set_lines(M.buf, 0, -1, true, string_lines)
+  vim.api.nvim_buf_set_option(M.buf, "modifiable", false)
+
+  for _, vl in ipairs(highlights) do
+    if vl.highlight then
+      vim.api.nvim_buf_add_highlight(M.buf, namespace, vl.highlight, vl.index, 0, -1)
+    end
+  end
+
+  for binding, cb in pairs(tree_keymaps) do
+    vim.keymap.set("n", binding, function()
+      local current_line = vim.api.nvim_win_get_cursor(0)[1]
+      local line = lines[current_line]
+      if line.node_ref ~= nil then
+        cb(line.node_ref)
+      end
+    end, { silent = true, noremap = true, buffer = M.buf })
+  end
+  for binding, cb in pairs(keymaps) do
+    vim.keymap.set("n", binding, function()
+      cb(M)
+    end, { silent = true, noremap = true, buffer = M.buf })
+  end
+end
+
+M.refresh = function()
+  print_lines()
+end
+
+local workflow_limit = 100
+
+local function get_workflow_runs_sync(co)
+  local lines = {}
+  vim.fn.jobstart(
+    "gh run list --json conclusion,displayTitle,event,headBranch,name,number,status,updatedAt,databaseId -L "
+      .. workflow_limit,
+    {
+      stdout_buffered = true,
+      on_stdout = function(_, data)
+        local json = vim.fn.json_decode(table.concat(data))
+        for _, value in ipairs(json) do
+          local status = value.status == "queued" and icons.pending
+            or value.status == "in_progress" and icons.in_progress
+            or value.conclusion == "failure" and icons.failed
+            or icons.succeeded
+
+          local conclusion = value.conclusion == "skipped" and icons.skipped
+            or value.conclusion == "failure" and icons.failed
+            or ""
+
+          local wf_run = {
+            status = status,
+            title = value.displayTitle,
+            display = value.displayTitle .. " " .. conclusion,
+            value = value.databaseId,
+            branch = value.headBranch,
+            name = value.name,
+            age = utils.format_date(value.updatedAt),
+            id = value.databaseId,
+          }
+          table.insert(lines, wf_run)
+        end
+      end,
+      on_exit = function()
+        coroutine.resume(co)
+      end,
+    }
+  )
+  coroutine.yield()
+  return lines
+end
+
+local function render(selected)
+  local new_buf = vim.api.nvim_create_buf(true, true)
+  M.buf = new_buf
+  vim.api.nvim_set_current_buf(new_buf)
+  populate_preview_buffer(selected.id, new_buf)
+  vim.api.nvim_buf_set_name(new_buf, "" .. selected.id)
+end
+
+M.previewer = function(self, entry)
+  local id = entry.value.id
+  M.buf = self.state.bufnr
+  populate_preview_buffer(id, self.state.bufnr)
+end
+
+M.list = function()
+  vim.notify "Fetching workflow runs (this may take a while) ..."
+  local co = coroutine.running()
+  local wf_runs = get_workflow_runs_sync(co)
+
+  require("octo.pickers.telescope.provider").workflow_runs(wf_runs, "Workflow runs", render)
```

Comment:
Use require "octo.picker" instead. Feel free to have M.not_implemented for the other two pickers

Reply:
done